src/providers/postgres/raster/qgspostgresrasterprovider.cpp

```diff
@@ -310,15 +310,15 @@ bool QgsPostgresRasterProvider::readBlock( int bandNo, const QgsRectangle &viewE
         const int _int { val.toInt( &ok ) };
         if ( ! ok )
         {
-          QgsMessageLog::logMessage( tr( "Cannot convert identified value to unsigned long" ), QStringLiteral( "PostGIS" ), Qgis::Warning );
+          QgsMessageLog::logMessage( tr( "Cannot convert identified value to int" ), QStringLiteral( "PostGIS" ), Qgis::Warning );
           return false;
         }
         std::memcpy( data, &_int, sizeof( int ) );
         break;
       }
       case Qgis::DataType::Int32:
       {
-        const long _long { val.toInt( &ok ) };
+        const long _long { val.toLong( &ok ) };
```

Comment:
seeing this "_long" identifier, I believe it is not a good idea. https://en.cppreference.com/w/cpp/language/identifiers mentions "identifiers that begin with an underscore are reserved in the global namespace".

Reply:
ok, I'll fix it in separate PR